src/main/kotlin/org/malachite/estella/services/SecurityService.kt

```diff
@@ -50,29 +50,37 @@ class SecurityService(
             .let { Base64.getEncoder().encode(it) }
             .let { String(it) }
 
-    private fun decryptDevPassword(password: String) =
-        password.toByteArray()
-            .let { Base64.getDecoder().decode(it) }
-            .let { String(it) }
-            .split(":")
-
+    private fun decryptDevPassword(password: String): Pair<UUID, UUID>? {
+        val passwordParts: List<UUID> = try {
+            Base64.getDecoder()
+                    .decode(password.toByteArray())
+                    .decodeToString()
+                    .split(":")
+                    .map { UUID.fromString(it)!! }
+        } catch (ex: Exception) {
+            Collections.emptyList()
+        }
 
-    fun compareOrganizationWithPassword(organization: Organization, password: String): Boolean {
-        val decrypted = decryptDevPassword(password)
-        if (decrypted.size != 2) throw UnauthenticatedException()
-        return decrypted[1].let {
-            taskStageRepository.findById(UUID.fromString(it))
-        }.let {
-            decrypted[0] == organization.id!!.toString() && it.isPresent
+        return when (passwordParts.size) {
+            2    -> Pair(passwordParts[0], passwordParts[1])
+            else -> null
```

Comment:
Po co tu zwracać nulla, skoro wiemy, że wtedy na pewno jest hasło złe i od razu można unatha rzucić

Reply:
Bo korzysta z tego później funkcja, która zwraca boola. Zresztą rzucanie unautha z funkcji, która tak naprawdę służy do parsowania jest dziwne

Reply:
Yyy... funkcja toUUID też rzuca wyjątek. Nie wiem, czemu nie moglibyśmy rzucić wyjątku z najwcześniejszego miejsca, które wykryje błędne hasło, jako ze i tak obsłużymy go handlerem, a nie złapiemy wyżej. Plus nawet jeśli z tej pary korzysta funkcja wyżej, to warunki w tej funkcji się uproszczą z uwagi na brak null checka. Myślę, że postępowanie według zasady "rzucaj wcześnie, łap późno" nie jest dziwne.